### PR TITLE
Adding few more exception types as user failure during downloading artifacts

### PR DESCRIPTION
**Pull request by GitHubSriramB**

### CODE REVIEW

File: src/Agent.Worker/Release/ReleaseJobExtension.cs [hunk 1]
```diff
@@ -261,19 +254,8 @@ private void CleanUpArtifactsFolder(IExecutionContext executionContext, string a
...
-                            IOUtil.DeleteDirectory(artifactsWorkingFolder, executionContext.CancellationToken);
-                        }
-
-                        Directory.CreateDirectory(artifactsWorkingFolder);
-                    }
-                    catch (Exception ex) when (ex is DirectoryNotFoundException || ex is UnauthorizedAccessException)
-                    {
-                        throw new ArtifactCleanupFailedException(StringUtil.Loc("FailedCleaningupRMArtifactDirectory", artifactsWorkingFolder), ex);
-                    }
+                    var releaseFileSystemManager = HostContext.GetService<IReleaseFileSystemManager>();
```
Comment (omeshp) [inline src/Agent.Worker/Release/ReleaseJobExtension.cs]: rename method to CreateArtifactsFolder?

File: src/Agent.Worker/Release/ReleaseJobExtension.cs [hunk 2]
```diff
@@ -261,19 +254,8 @@ private void CleanUpArtifactsFolder(IExecutionContext executionContext, string a
                 {
                     executionContext.Output(StringUtil.Loc("RMCleaningArtifactsDirectory", artifactsWorkingFolder));
```
Comment (omeshp) [inline src/Agent.Worker/Release/ReleaseJobExtension.cs]: update messages?

File: src/Agent.Worker/Release/ReleaseJobExtension.cs [hunk 3]
```diff
@@ -261,19 +254,8 @@ private void CleanUpArtifactsFolder(IExecutionContext executionContext, string a
...
 
-                    try
-                    {
-                        if (Directory.Exists(artifactsWorkingFolder))
```
Comment (omeshp) [inline src/Agent.Worker/Release/ReleaseJobExtension.cs]: will removing delete directory here impact download of custom task based artifacts?

File: src/Agent.Worker/Release/ReleaseJobExtension.cs (hunk 1 above)
Comment (GitHubSriramB) [inline src/Agent.Worker/Release/ReleaseJobExtension.cs]: done

**Pull request merged**